day04/lib/main.rb

```diff
@@ -0,0 +1,114 @@
+# => TODO
+# => warehouse (how many product we have avaible)
+# => total price in basket + tax
+
+class Main
+  require_relative "./product"
+  require_relative "./warehouse"
+  require_relative "./basket"
+
+  attr_reader :action
+  def initialize
+    get_basket
+    get_list
+  end
+
+  def get_list
+    @get_list ||= Warehouse.new.make_list
+  end
+
+  def get_basket
+    bas = Basket.new
+    @basket ||= bas.create_basket
+  end
+
+  def choose_product(option = "add")
+    show_products
+    puts "Choose a product"
+    product = gets.chomp
+    puts "Your choice is: " + product
+    if get_product_name.include?(product)
+      if option == "add"
+        return get_product_from_warehouse(product)
```

Comment:
I think that you don't need `return` here. Ruby by default returns last thing that it gets from calling a method.